### PR TITLE
onagre: 1.0.0-alpha.0 -> 1.1.0

### PR DESCRIPTION
**Pull request by ilya-epifanov**
## Description of changes

https://github.com/onagre-launcher/onagre/blob/main/CHANGELOG.md#100---2024-02-14

## Things done

Added xorg and wayland libraries as runtime dependencies.
Updated to 1.0.0 from the upstream.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 296846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by jfvillablanca** [comment]
> Result of `nixpkgs-review pr 296846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package built:

You accidentally removed `pop-launcher` which is a runtime requirement

```
❯ ./results/onagre/bin/onagre
2024-03-19T11:53:15.136749Z  INFO onagre: Starting onagre
2024-03-19T11:53:15.136822Z ERROR onagre::app::style: Failed to parse theme "/home/user/.config/onagre/theme.scss": Failed to open config file
2024-03-19T11:53:15.136831Z  WARN onagre::app::style: Failing back to default theme
2024-03-19T11:53:15.164041Z ERROR onagre::app::subscriptions::pop_launcher: Failed to start pop-launcher backend.
2024-03-19T11:53:15.164052Z ERROR onagre::app::subscriptions::pop_launcher: Make sure either pop-launcher or onagre-launcher is installed.
2024-03-19T11:53:15.164057Z ERROR onagre::app::subscriptions::pop_launcher: See: https://github.com/pop-os/launcher or https://github.com/onagre-launcher/launcher
```

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 296846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by ilya-epifanov** [comment]
So I've been using it for a couple of days and everything works just fine

**Comment by jfvillablanca** [comment]
> So I've been using it for a couple of days and everything works just fine

It's not reproducible on my machine. I ran the review again and attempted to run both wrapped and unwrapped binaries. There must be a runtime dependency that you have system wide that's not included in the expression.

I'm using the i3 window manager so I lack some things usually provided by a desktop environment.

```
nixpkgs-review/pr-296846-2  shell
❯ ./results/onagre/bin/onagre
2024-03-23T13:43:55.760481Z  INFO onagre: Starting onagre
2024-03-23T13:43:55.760551Z ERROR onagre::app::style: Failed to parse theme "/home/user/.config/onagre/theme.scss": Failed to open config file
2024-03-23T13:43:55.760560Z  WARN onagre::app::style: Failing back to default theme
thread 'main' panicked at src/app/mod.rs:539:33:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
nixpkgs-review/pr-296846-2  shell
❯ ./results/onagre/bin/.onagre-wrapped
2024-03-23T13:44:03.789624Z  INFO onagre: Starting onagre
2024-03-23T13:44:03.789699Z ERROR onagre::app::style: Failed to parse theme "/home/user/.config/onagre/theme.scss": Failed to open config file
2024-03-23T13:44:03.789708Z  WARN onagre::app::style: Failing back to default theme
2024-03-23T13:44:03.810716Z ERROR onagre::app::subscriptions::pop_launcher: Failed to start pop-launcher backend.
2024-03-23T13:44:03.810730Z ERROR onagre::app::subscriptions::pop_launcher: Make sure either pop-launcher or onagre-launcher is installed.
2024-03-23T13:44:03.810734Z ERROR onagre::app::subscriptions::pop_launcher: See: https://github.com/pop-os/launcher or https://github.com/onagre-launcher/launcher
```

**Comment by ilya-epifanov** [comment]
@jfvillablanca please test again against the latest commit. It should work properly but if it isn't please guide me on how to correct that.

**Comment by jfvillablanca** [comment]
> @jfvillablanca please test again against the latest commit. It should work properly but if it isn't please guide me on how to correct that.

I tried to run this both on X (i3) and Wayland (Hyprland). It works fine in Wayland so we might be missing some X dependency

**Comment by teto** [comment]
works here on sway (though it runs quite sluggish). I have on startup
```
2024-04-03T09:36:08.780155Z  INFO onagre: Starting onagre
2024-04-03T09:36:08.780256Z ERROR onagre::app::style: Failed to parse theme "/home/teto/.config/onagre/theme.scss": Failed to open config file
2024-04-03T09:36:08.780268Z  WARN onagre::app::style: Failing back to default theme
```
The border is black since it doesn't seem to load any fallback scss. Would it be possible to add the official default scss (I assume there is one) in the derivation so that one can either copy/paste it or reference the theme.scss from the derivation (via home-manager for instance) ?

**Comment by ilya-epifanov** [comment]
@jfvillablanca could you please check with the latest  commit?
Also, if it fails, could you attach the output of `RUST_BACKTRACE=full RUST_LOG=info onagre` ?

**Comment by ilya-epifanov** [comment]
@jfvillablanca let's revive this PR. I'm using it successfully for more than a month already. Tested with wayland and Xwayland.

**Review by jfvillablanca — APPROVED** [review]
Works on X and Wayland. Tried in i3 and Hyprland

**Comment by ilya-epifanov** [comment]
@jfvillablanca it's my first PR, so excuse a silly question. How do we merge it?

**Comment by jfvillablanca** [comment]
> @jfvillablanca it's my first PR, so excuse a silly question. How do we merge it?

If I'm not mistaken, we just wait for a `nixpkgs-committer` to merge it. I believe they do it in batches. This PR has been labeled with the approvals label. It should be merged in a bit :smile:

File: pkgs/applications/misc/onagre/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/onagre/default.nix:28]:
```suggestion
  cargoHash = "sha256-xC5nf7DJCMBR2gUoB5NWZJGuZ8brs8WklCzg/osCsho=";
```

File: pkgs/applications/misc/onagre/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/onagre/default.nix:31]: This looks weird. Just write them in separate lines.

File: pkgs/applications/misc/onagre/default.nix
Comment (Aleksanaa) [inline pkgs/applications/misc/onagre/default.nix:23]:
```suggestion
    repo = "onagre";
```

**Comment by Aleksanaa** [comment]
Diff too big, please move file in another commit (otherwise GitHub won't track it correctly)

**Comment by ilya-epifanov** [comment]
@Aleksanaa @jfvillablanca so, I've updated it to 1.1.0 now that it's out. I've also moved the package to `pkgs/by-name` as it's a new requirement.

**Comment by jfvillablanca** [comment]
Result of `nixpkgs-review pr 296846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
Please reword commit to `onagre: moved to pkgs/by-name`

**Comment by teto** [comment]
Result of `nixpkgs-review pr 296846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>onagre</li>
  </ul>
</details>

**Pull request merged**
